### ISSUE TITLE
Markdown only view

### ISSUE THREAD

**Issue by ho1ger** [nextcloud/text#123]
It would be quite useful to have a plain text view that only shows the markdown.

**Comment by jancborchardt** [comment]
Not sure if we’ll do this – @juliushaertl?

There are already 2 apps which do a good job at this, being the [Markdown Editor](https://apps.nextcloud.com/apps/files_markdown) and the [Notes app](https://apps.nextcloud.com/apps/notes). :)

**Comment by ho1ger** [comment]
hmmmm, ... 

When my understanding is correct, that this app becomes the default text editor in Nextcloud, it would be quite useful to be able to switch to plain text mode. Let me give an example: There might be text files that contain characters that the parser mistakenly parses as markdown. For instance, you have a text file which maybe contains some bash scripts examples and in the bash there's a line with a comment like this:

# comment bla bla... 

This would be rendered as a nice headline ;-)

**Comment by ho1ger** [comment]
Nice :D I actually didn't want that my example

`# comment bla bla...`

would be rendered...

**Comment by lachmanfrantisek** [comment]
@ho1ger You can use code block for that specific purpose:

```
# comment bla bla...
cat *.txt
```
(Start and end the block with ```)

**Comment by jancborchardt** [comment]
> Let me give an example: There might be text files that contain characters that the parser mistakenly parses as markdown. For instance, you have a text file which maybe contains some bash scripts examples and in the bash there's a line with a comment

In that case this issue is a duplicate of **Support plain text files #13** and will be taken care of. :)

**Comment by heavyelement** [comment]
This is also a feature I'd like to see. I'd love to be able to switch between a plaintext view and a styled view. If it was a keystroke, that would be excellent!

**Comment by juliushaertl** [comment]
We could open markdown files in plain text mode (after #129) but that would limit all editors to plain text editing as well as not allowing any rich document editing features that tiptap has.

**Comment by rkaraba** [comment]
Best way to do this is to have button witch allows you to switch between  plaintext view and a markdown editor.
Something like in current text editor. 
You can view plain text and markdown at the same time or only plain text or only markdown
Of course there is no possibility to edit markdown in current text editor

![obrázok](https://user-images.githubusercontent.com/20682075/60455499-d196d180-9c36-11e9-806d-bc4cdbe2c4a2.png)

**Comment by juliushaertl** [comment]
I have been thinking a bit more about this and unfortunately I don't see a way to bring this together with the collaborative editing. Prosemirror uses a document model that in our case either supports plaintext or rich editing, but not both at the same time. We have proper plaintext support for non-markdown files with #129 and rich editing for markdown files, which should cover most of the use cases. The markdown editor will still continue to work like before and can be used besides the text app. 

Since we aimed for a rich editor this is out of scope for the text app.

**Comment by spitzlei** [comment]
https://github.com/nextcloud/text/issues/123#issuecomment-507358050

This would be great. Only reason for not using the editor in our company is, that we can't use md/plain at same time. Without this option the editor is not usable for us. Maybe it can be reconsidered to implement this?

**Comment by juliushaertl** [comment]
As mentioned, this is a technical limitation of the rich editor, so this is most-likely not going to happen. However the files_markdown editor should continue to work just fine, if you really need the two-sided layout.

**Comment by KimmoJ** [comment]
Not having the ability to edit, or even see, the markdown straight in the editor is a fail. One of the points of markdown is that it's easy to read and edit even in source form. What a shame.

**Comment by jancborchardt** [comment]
@KimmoJ as mentioned there’s the [Markdown Editor](https://apps.nextcloud.com/apps/files_markdown) and the [Notes app](https://apps.nextcloud.com/apps/notes) already if you need that.

The primary goal of the Text app is to make quick rich-text editing and collaboration work for the masses, not only people who know what Markdown is and how to read it. Markdown being "easy to read in source form" might be true for programmers and people who are used to it, but Nextcloud doesn’t only cater to those people.

**Comment by severin-lemaignan** [comment]
Not having the possibility to see and edit the Markdown source is indeed an issue for us.

@juliushaertl if the editor can not do collaborative editing for both rich text and plain text at the same time, why not having a simple switch to either view the file as rich text or as plain text.

**Comment by dreamerworks** [comment]
+1 for text mode. Acceptable even as keyboard shortcut. Ideally a toggle button.

**Comment by genofire** [comment]
@juliushaertl  sorry for writing on closed issues.
I just try to understand the technical limits.
Is the markdown rendered server-side?
(if not, i do not understand, where is the technical problem. it should be possible to have different users has different live views on the same current file)

**Comment by dpprdan** [comment]
@genofire https://github.com/nextcloud/text/issues/323#issuecomment-538461645

**Comment by claell** [comment]
I guess according to https://github.com/nextcloud/text/issues/123#issuecomment-508723049 and what I have learned from other comments, the problem is that there are two document formats generated when using WYSIWYG mode and thus offering to edit both at the same time can lead to inconsistencies as then there is no single source of truth anymore.

I suggest overthinking the current system. Maybe there is a way to always treat the Markdown document as "truth". Maybe one does not have to derive a different document format from it to allow WYSIWYG editing in the first place.

**Comment by Sieboldianus** [comment]
The Markdown Editor is not compatible with Nextcloud 22 (because Plaintext Editor, which is needed, is not compatible). :(

**Comment by bentuna** [comment]
Things are getting especially impractical when you try to use parts of a markdown file by copying it out of Nextcloud text and pasting it into a Talk message: All formatting and links are gone - although talk supports markdown.

**Comment by jancborchardt** [comment]
@bentuna that sounds like a different issue (and one that should definitely be fixed) – could you open a separate issue about it?

**Comment by hbarel** [comment]
I'd be very interested to know if anyone on this thread has devised a way to just edit plain text files on NC.
The Markdown editor is sometimes too little and sometimes too much...
